crates/ruff/resources/test/fixtures/tryceratops/TRY302.py

```diff
@@ -0,0 +1,111 @@
+"""
+Violation:
+
+Checks for uses of `raise` directly after a `rescue`.
+"""
+class MyException(Exception):
+    pass
+
+def bad():
+    try:
+        process()
+    except Exception:
+        raise
+
+def bad():
+    try:
+        process()
+    except Exception:
+        raise
+        print("this code is pointless!")
+
+def bad():
+    try:
+        process()
+    except:
+        # I am a comment, not a statement!
+        raise
+
+def bad():
+    try:
+        process()
+    except Exception:
+        raise
+
+def bad():
+    try:
+        process()
+    except Exception as e:
+        raise
+
+def bad():
+    try:
+        process()
+    except Exception as e:
+        raise e
+
+def bad():
+    try:
+        process()
+    except MyException:
+        raise
+    except Exception:
+        raise
+
+def bad():
+    try:
+        process()
+    except MyException as e:
+        raise e
+    except Exception as e:
+        raise e
+
+def bad():
+    try:
+        process()
+    except MyException as ex:
+        raise ex
+    except Exception as e:
+        raise e
+
+def fine():
+    try:
+        process()
+    except Exception as e:
+        raise ex
```

Comment:
Does that even compile? Catch e, but raise ex?

Reply:
You would get a runtime error, since `ex` can't be resolved. But it wouldn't trigger TRY302.

Reply:
Yeah it's intentionally invalid. It won't compile but also shouldn't trigger the lint